src/transactions/TransactionFrame.cpp

```diff
@@ -645,7 +670,7 @@ TransactionFrame::validateSorobanResources(SorobanNetworkConfig const& config,
     auto const& writeEntries = resources.footprint.readWrite;
     if (resources.instructions > config.txMaxInstructions())
     {
-        pushSimpleDiagnosticError(
+        pushValidationTimeDiagnosticError(
```

Comment:
If config were to change after a transaction has been accepted to the mempool, that transaction would still not make it to apply time because if would fail validation when the transaction set is created right? This assumes the config was reduced to something that would invalidate the transactions.

Reply:
Hmm, I guess yes, we would attach the diagnostics to the transaction in this scenario, even though we don't return the result back to the user. But this seems like a very rare scenario, so probably it's negligible. The invariant of not attaching the diagnostics to meta will not be invalidated either (FWIW tx sets with invalid transactions can't be applied).

Reply:
> FWIW tx sets with invalid transactions can't be applied

Yeah that's what I wanted to confirm. This sounds fine to me then.